uiowa/uiowa
# customizable event categories

Resolves #6364 
~~- [x] Update hook to create the "Event categories" taxonomy if it does not exist.~~
- [x] Add the Event categories field to the Event content type and assign the taxonomy to it.
- [x] Decide on a place for the field to go. In the main area of the form? in the relationships dropdown? somewhere new?
- [x] Hook to move relevant terms to the event categories taxonomy (eg: The "event_type" terms from ISWF to "Event categories").
    ~~- [x] This also includes reworking how events reference genres I think?~~
- [x] Set perms for webmaster/publisher to have add/edit/delete permissions.
- [x] remove splits from ISWF and Admissions to consolidate implementations in the event module
- [x] Ensure that admissions and ISWF still function in the desired way after removal of split functionality.
~~- [x] Should we allow for the renaming of the event categories field? (eg: ISWF might want it to be called "Event type")~~
- [x] display the event categories at the bottom of events
    - [x] This means removing the complete splits for displays for admissions and iowasummerwritingfestival and adding it as a partial and ensuring that the pages still function the intended way.
- [x] Set Perms

After all this is done, new terms should be able to be added/edited by webmasters/publishers.

## To Test

Gremlin check before sync
```
ddev drush @iowasummerwritingfestival.local sql-drop -y && ddev drush @iowasummerwritingfestival.local cr && ddev blt ds --site=iowasummerwritingfestival.uiowa.edu
```

1. Pull main and observe that ISWF has an event type field and it is populated with many options.
2. Pull this branch and do all the setup stuff.
3. Sync ISWF and observe that a taxonomy has been created called 'event_categories' with all the options previously available in 'event_type'
4. Observe that events have the 'event_categories' field and that the 'event_type' data has been migrated in to it properly. You can use ISWF's live site to check what they were previously.
5. Observe that there are only slight visible changes to ISWF site. Event categories should be the only change, and the events display should remain roughly the same.
6. Sync Oniowa and Admissions
7. Observe that they do not change from their live version.
8. ???
9. Profit.
